detect.go

```diff
@@ -0,0 +1,59 @@
+package enable
+
+import (
+	"fmt"
+	"path/filepath"
+
+	"os"
+
+	"github.com/paketo-buildpacks/packit/v2"
+
+	"github.com/paketo-buildpacks/packit/v2/fs"
+	"github.com/paketo-buildpacks/packit/v2/scribe"
+)
+
+var nodeRequirement = packit.BuildPlanRequirement{
+	Name: "node",
+	Metadata: map[string]interface{}{
+		"launch": true,
+	},
+}
+
+var nodeModulesRequirement = packit.BuildPlanRequirement{
+	Name: "node_modules",
+	Metadata: map[string]interface{}{
+		"launch": true,
+	},
+}
+
+func Detect(logger scribe.Emitter) packit.DetectFunc {
+	return func(context packit.DetectContext) (packit.DetectResult, error) {
+		logger.Debug.Title("%s %s", context.BuildpackInfo.ID, context.BuildpackInfo.Version)
+
+		if os.Getenv("BP_INCLUDE_NODEJS_RUNTIME") != "true" {
+			return packit.DetectResult{}, packit.Fail.WithMessage("BP_INCLUDE_NODEJS_RUNTIME is not set to true")
+		}
+		logger.Debug.Process("NodeJS is wanted for runtime")
+
+		buildPlan := packit.BuildPlan{
+			Requires: []packit.BuildPlanRequirement{nodeRequirement},
+		}
+
+		hasPackageJSON, err := fs.Exists(filepath.Join(context.WorkingDir, "package.json"))
+		if err != nil {
+			return packit.DetectResult{}, fmt.Errorf("failed to check for package.json: %w", err)
+		}
+
+		if hasPackageJSON {
+			logger.Debug.Process("node_modules is wanted for runtime")
+			buildPlan.Requires = []packit.BuildPlanRequirement{nodeRequirement, nodeModulesRequirement}
+			buildPlan.Or = []packit.BuildPlan{{
```

Comment:
Do I understand this correctly? It basically means: if the `nodeModulesRequirement` was not provided by some other buildpack, we are ok with just having the`nodeRequirement`?

Reply:
I understand it as well like that.

From the Or code:

```
// Or is a list of additional BuildPlans that may be selected by the
// lifecycle
```